front/pages/api/w/[wId]/data_sources/[dsId]/files.ts

```diff
@@ -124,4 +119,119 @@ async function handler(
   }
 }
 
+function handlePublicErrorResponse(
+  req: any,
+  res: any,
+  error: {
+    name: string;
+    code:
+      | "internal_server_error"
+      | "invalid_request_error"
+      | "file_too_large"
+      | "file_type_not_supported"
+      | "unauthorized"
+      | "quota_exceeded"
+      | "resource_not_found"
+      | "processing_failed"
+      | "validation_error";
+    message: string;
+  }
+) {
+  switch (error.code) {
+    case "validation_error":
+    case "invalid_request_error":
+      logger.error(
```

Comment:
No need for these specific logger. You can have maybe one generic but no need for that much specificity